### PR TITLE
HY_PD+MyEvents_query_emailService

### PR DESCRIPTION
**Pull request by 2HEADSs**
1. add functionality for "myEventsForApproval", "myUpcomingEvents" and "myPastEvents" to work with query (region, category, search)
2. update emailService to send emails when user/users or event is approved/disapproved

### CODE REVIEW

File: src/controllers/swaggerUserDocs.js [hunk 1]
```diff
@@ -317,7 +317,8 @@
  *   put:
  *     summary: Approve/Disapprove user.
  *     description: Approve/Disapprove user `isApproved` property by their ID.
- *      Only active admin can change `isApproved` field.
+ *      Only active admin can change `isApproved` field. After a successful operation,
+ *      emails will be sent to the users.
```
[the remark this answers is not in the document]
Comment (2HEADSs) [reply src/controllers/swaggerUserDocs.js]: fixed

File: src/controllers/swaggerUserDocs.js [hunk 2]
```diff
@@ -349,7 +350,8 @@
  *   put:
  *     summary: Approve/Disapprove users.
  *     description: Update users `isApproved` property by their ID.
- *      Only active admin can change `isApproved` field.
+ *      Only active admin can change `isApproved` field. After a successful operation,
+ *      emails will be sent to the users.
```
[the remark this answers is not in the document]
Comment (2HEADSs) [reply src/controllers/swaggerUserDocs.js]: fixed

File: src/services/emailService.js [hunk 3]
```diff
@@ -42,7 +42,84 @@ async function resetPassword(requestBody) {
   // console.log('Email sent: ' + info.response);
   return info;
 }
+//TODO: update reamde
+async function sendWhenApproveDisapproveUsers(usersList, isApproved) {
+  let text = 'Your profile has been approved in Race Fanatic app.';
```
[the remark this answers is not in the document]
Comment (2HEADSs) [reply src/services/emailService.js]: fixed

File: src/services/emailService.js [hunk 4]
```diff
@@ -42,7 +42,84 @@ async function resetPassword(requestBody) {
...
+  // console.log('Email sent: ' + info.response);
+  return info;
+}
+//TODO: update reamde
+async function sendWhenApproveDisapproveEvent(
+  userEmail,
+  isApproved,
+  eventName
+) {
+  let text =
```
[the remark this answers is not in the document]
Comment (2HEADSs) [reply src/services/emailService.js]: fixed

File: src/services/emailService.js [hunk 5]
```diff
@@ -42,7 +42,84 @@ async function resetPassword(requestBody) {
   // console.log('Email sent: ' + info.response);
   return info;
 }
+//TODO: update reamde
+async function sendWhenApproveDisapproveUsers(usersList, isApproved) {
```
[the remark this answers is not in the document]
Comment (2HEADSs) [reply src/services/emailService.js]: fixed

File: src/services/emailService.js [hunk 6]
```diff
@@ -42,7 +42,84 @@ async function resetPassword(requestBody) {
...
+    text: text,
+  };
+
+  const info = await transporter.sendMail(mailOptions);
+
+  // console.log('Email sent: ' + info.response);
+  return info;
+}
+//TODO: update reamde
+async function sendWhenApproveDisapproveEvent(
```
[the remark this answers is not in the document]
Comment (2HEADSs) [reply src/services/emailService.js]: fixed

**Review by Getsov — APPROVED** [review]

**Pull request merged**